perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -1093,7 +1093,22 @@ public void activateGroupResourceAssignment(PerunSession sess, Group group, Reso
 	@Override
 	public void processGroupResourceActivationAsync(PerunSession sess, Group group, Resource resource) {
 		try {
-			processGroupResourceActivation(sess, group, resource);
+			// try activation for 5 times after 5 seconds because the assignment
+			// might not be committed in DB yet
+			boolean successful = false;
+			for (int i = 0; i < 5; ++i) {
+				try {
+					processGroupResourceActivation(sess, group, resource);
+					successful = true;
+					break;
+				} catch (GroupNotDefinedOnResourceException ex) {
+					Thread.sleep(5000);
+				}
+			}
+
+			if (!successful) {
+				processGroupResourceActivation(sess, group, resource);
+			}
```

Comment:
Why is here this extra activation? Is it somehow different from the ones in the for-cycle? This way, if the exception was not GroupNotDefinedOnResourceException, it will try to activate it again, which is not something we want to do (at least I think).

Reply:
If the exception was not GroupNotDefinedOnResourceException, it will be catched in outer try-catch block. The extra activation is there because if GroupNotDefinedOnResourceException was catched 5 times in for cycle, it will be now catched in outer try-catch block, logged and FAILED status will be set.